src/ADAL.PCL/AcquireTokenHandlerBase.cs

```diff
@@ -169,17 +173,21 @@ internal abstract class AcquireTokenHandlerBase
                             this.NotifyBeforeAccessCache();
                             notifiedBeforeAccessCache = true;
                         }
-
                         this.tokenCache.StoreToCache(ResultEx, this.Authenticator.Authority, this.Resource, this.ClientKey.ClientId, this.TokenSubjectType, this.CallState);
                     }
                 }
-
                 await this.PostRunAsync(ResultEx.Result);
                 return ResultEx.Result;
             }
             catch (Exception ex)
             {
+                
                 PlatformPlugin.Logger.Error(this.CallState, ex);
+                if (client!=null && client.Resiliency && extendedLifetimeResultEx != null)
+                {
+                    PlatformPlugin.Logger.Information(this.CallState, "Refreshing AT failed either due to one of these :- Internal Server Error,Gateway Timeout and Service Unavailable.Hence returning back stale AT as a part of client resiliency");
```

Comment:
don't use internal terms in logging like "client resiliency". The error should already be logged. if we add more conditions then  you would need to come back and update this log message as well.